lib/post_revisor.rb

```diff
@@ -330,7 +330,8 @@ def revise!(editor, fields, opts = {})
     QuotedPost.extract_from(@post)
     TopicLink.extract_from(@post)
 
-    Topic.reset_highest(@topic.id)
+    # Skip resetting highest post number if only changing post ownership
+    Topic.reset_highest(@topic.id) unless @fields.keys.size == 1 && @fields.has_key?("user_id")
```

Comment:
Small nitpick. Accessing `@fields.keys` allocates a new `Array` which is unnecessary if you just want to know how many items the `Hash` contains. You can simplify it:

```suggestion
    Topic.reset_highest(@topic.id) unless @fields.size == 1 && @fields.has_key?("user_id")
```

or

```suggestion
    Topic.reset_highest(@topic.id) unless @fields.one? && @fields.has_key?("user_id")
```